exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/MergeTableTest.kt

```diff
@@ -324,4 +326,139 @@ class MergeTableTest : MergeBaseTest() {
+            override val primaryKey: PrimaryKey = PrimaryKey(id)
+        }
+
+        withTables(excludeSettings = TestDB.ALL - TestDB.ALL_POSTGRES, source, dest) {
+            val id = source.insertAndGetId { }
+
+            dest.mergeFrom(source) {
+                // `overridingSystemValue` allows to overwrite `generated always as identity` value
+                // otherwise Postgres will throw exception that auto generated value must not be overwritten
+                whenNotMatchedInsert(overridingSystemValue = true) {
+                    it[dest.id] = source.id
+                }
+            }
+
+            assertEquals(id.value, dest.selectAll().single()[dest.id].value)
+        }
+    }
+
+    @Test
+    fun testOverridingUserValue() {
+        val source = object : IntIdTable() {
+        }
+
+        val sequenceStartNumber = 100
+        val dest = object : IdTable<Int>() {
+            override val id =
+                integer("id").withDefinition("GENERATED BY DEFAULT AS IDENTITY(SEQUENCE NAME testOverridingUserValueSequence START WITH $sequenceStartNumber)")
+                    .entityId()
+            override val primaryKey: PrimaryKey = PrimaryKey(id)
+        }
+
+        withTables(excludeSettings = TestDB.ALL - TestDB.ALL_POSTGRES, source, dest) {
+            try {
+                source.insertAndGetId { }
+
+                dest.mergeFrom(source) {
+                    // `overridingUserValue` here allows to avoid setting id value from source table, and take generated by sequence instead
+                    whenNotMatchedInsert(overridingUserValue = true) {
+                        it[dest.id] = source.id
+                    }
+                }
+
+                assertEquals(sequenceStartNumber, dest.selectAll().single()[dest.id].value)
+            } finally {
+                SchemaUtils.drop(dest)
+                exec("drop sequence if exists testOverridingUserValueSequence")
+            }
+        }
+    }
+
+    @Test
+    fun testInsertDefaultValues() {
+        val source = object : IntIdTable("test_source") {
+            val value = varchar("value", 128)
+        }
+        val dest = object : IntIdTable("test_dest") {
+            // `withDefinition()` here is a workaround to avoid insert explicitly pass default value
+            val value = varchar("value", 128).withDefinition("DEFAULT", stringLiteral("default-test-value")).databaseGenerated()
+        }
+
+        withTables(excludeSettings = defaultExcludeSettings, source, dest) {
+            addLogger(StdOutSqlLogger)
+            source.insertAndGetId {
+                it[value] = "user-defined-value"
+            }
+
+            dest.mergeFrom(source) {
+                whenNotMatchedInsert {}
+            }
+
+            assertEquals("default-test-value", dest.selectAll().single()[dest.value])
+        }
+    }
+
+    @Test
+    fun testUnsupportedPostgresFeatures() {
```

Comment:
Really minor comment -> imo This name initially made it seem like it is testing features that are unsupported by PostgreSQL, not the other way around in other db.